### ISSUE TITLE
(Fine-tune) Train on my own dataset, but the loss values are not convergence.

### ISSUE THREAD

**Issue by keishatsai** [mrharicot/monodepth#107]
Hello, I just wondering that why the loss value became higher during batch 44000, and at the first time, it went well to convergence. Is it because of my batch_size is too small? Currently, I am using `batch_size = 4` to avoid the memory crash, and there are about 5500 samples on my own dataset.

`model-287200` is trained on cityscapes datasets with batch_size=4, and all the other settings remained the same. 
Here is my command line:
`$python monodepth_main.py --mode train --batch_size 4 --checkpoint_path ~/wonderland/monodepth-tmp/model20180130/model20180130/model-287200 --retrain --encoder resnet50 --model_name model20180206city2my--data_path ~/wonderland/ --filenames_file ~/wonderland/paper_code/monodepth-master/utils/filenames/my_train_files.txt --log_directory ~/wonderland/monodepth-tmp/model20180206 --full_summary`

total number of samples: 5500
total number of steps: 68750

batch    100 | examples/s: 8.42 | loss: 0.36922 | time elapsed: 0.02h | time left: 11.92h
batch    200 | examples/s: 8.26 | loss: 0.41477 | time elapsed: 0.03h | time left: 10.46h
batch    300 | examples/s: 8.14 | loss: 0.33602 | time elapsed: 0.04h | time left: 9.94h
batch    400 | examples/s: 8.66 | loss: 0.38731 | time elapsed: 0.06h | time left: 9.71h
batch    500 | examples/s: 8.64 | loss: 0.43357 | time elapsed: 0.07h | time left: 9.55h
batch   1000 | examples/s: 8.52 | loss: 0.34376 | time elapsed: 0.14h | time left: 9.21h
batch   2000 | examples/s: 8.13 | loss: 0.30938 | time elapsed: 0.27h | time left: 8.96h
batch   3000 | examples/s: 8.38 | loss: 0.41769 | time elapsed: 0.40h | time left: 8.82h
...
batch  44100 | examples/s: 9.16 | loss: 3.75232 | time elapsed: 5.82h | time left: 3.25h
batch  45300 | examples/s: 9.20 | loss: 1.19604 | time elapsed: 5.97h | time left: 3.09h
batch  45400 | examples/s: 9.20 | loss: 1.20178 | time elapsed: 5.98h | time left: 3.08h
batch  45500 | examples/s: 9.19 | loss: 1.47934 | time elapsed: 5.99h | time left: 3.06h
batch  45600 | examples/s: 8.76 | loss: 1.23994 | time elapsed: 6.01h | time left: 3.05h
batch  45700 | examples/s: 8.63 | loss: 1.43722 | time elapsed: 6.02h | time left: 3.04h
...
batch  68700 | examples/s: 8.99 | loss: 1.28485 | time elapsed: 8.89h | time left: 0.01h

`Another question is that I tried to add flag full_summary , and it seems to the summary doesn't have  been saved. `

Do you have any idea why I encounter these problems? 
Thank you.
Keisha

**Comment by mrharicot** [comment]
Hi,
I have seen this behaviour sometimes, especially on the stereo model.
Do the disparities look good before it fails?

**Comment by keishatsai** [comment]
Good point!! I haven't tried the checkpoint before batch 44000 because I thought this model wasn't trained well.  Let me try, and I will put the result here later.

**Comment by keishatsai** [comment]
![frame-003325_disp_pre](https://user-images.githubusercontent.com/8175070/36075117-e2bdb57c-0f84-11e8-96f7-ef053ca670f6.png)

**Comment by keishatsai** [comment]
`python monodepth_simple.py --image_path ~/wonderland/monodepth-test-tmp/model20180206/img_5039.png --encoder resnet50 --checkpoint_path ~/wonderland/monodepth-tmp/model20180206/model-40000`
I used NYU dataset to test this model.
My test result:
![img_5039_disp_pre](https://user-images.githubusercontent.com/8175070/36075269-7dbaa060-0f87-11e8-9b6e-b8b6af87a7fa.png)
Original Input:
![img_5039](https://user-images.githubusercontent.com/8175070/36075290-bfac9e56-0f87-11e8-985f-06409539d382.png)

**Comment by mrharicot** [comment]
Hi,
Did you fine tune on a similar dataset?
If you trained only on Cityscapes you can't really expect to do great on NYU.

**Comment by keishatsai** [comment]
Hello~~ Yes, I did train on similar dataset (my own indoor dataset).
Other results are the same after testing on the same checkpoint.

The point is that I trained cityscape dataset with batch size 4 and then fine tune the model by my own indoor dataset(which the loss is not convergence). However, I have also tried to fine tune your pretrain model_cityscape_resnet by my own dataset. The loss result wasn't diverse, and I have no idea why. Is it because the batch size?

**Comment by mrharicot** [comment]
Hi,
Can you share a couple of input stereo frames?

**Comment by keishatsai** [comment]
Hi, did you mean the training dataset? If so,  I am afraid that I can’t do that because it is private area XD.  Do you have any idea that the divergence may occur due to other factors besides training dataset? I trained the same dataset of Cityscape in resnet(batch size 4), and then I tried to fine tune it with my own dataset. The result was diverse. I also fine tune your pretrained model with my own dataset (batch size 4), but the result is convergence.

**Comment by Usernamezhx** [comment]
@keishatsai  hi you said 

>  I also fine tune your pretrained model with my own dataset (batch size 4), but the result is convergence
means that the **model_kitti: Our main model trained on the kitti split** model？？？
